Core/include/Acts/Propagator/detail/GenericDenseEnvironmentExtension.hpp

```diff
@@ -175,12 +175,14 @@ struct GenericDenseEnvironmentExtension {
         g / (newMomentum * newMomentum * newMomentum);
 
     // Update momentum
-    state.stepping.p = newMomentum;
+    state.stepping.pars[eFreeQOverP] =
+        std::copysign(1. / newMomentum, state.stepping.pars[eFreeQOverP]);
```

Comment:
This line suggests that the charge is -1,0,1. If this is the (reasonable) way to go, then the charge could even become a bool. The sign can be obtained from the `pars[eFreeQOverP]`

Reply:
Sorry, this will be updated to use the charge ...

Reply:
Updated.